sentry-android-fragment/src/main/java/io/sentry/android/fragment/FragmentLifecycleIntegration.kt

```diff
@@ -0,0 +1,66 @@
+package io.sentry.android.fragment
+
+import android.app.Activity
+import android.app.Application
+import android.app.Application.ActivityLifecycleCallbacks
+import android.os.Bundle
+import androidx.fragment.app.FragmentActivity
+import io.sentry.IHub
+import io.sentry.ILogger
+import io.sentry.Integration
+import io.sentry.SentryLevel.DEBUG
+import io.sentry.SentryOptions
+import java.io.Closeable
+
+class FragmentLifecycleIntegration(
+    private val application: Application,
+    private val logger: ILogger
+) : ActivityLifecycleCallbacks, Integration, Closeable {
+
+    private lateinit var hub: IHub
+
+    override fun register(hub: IHub, options: SentryOptions) {
+        this.hub = hub
+
+        application.registerActivityLifecycleCallbacks(this)
+        logger.log(DEBUG, "FragmentLifecycleIntegration installed.")
+    }
+
+    override fun close() {
+        application.unregisterActivityLifecycleCallbacks(this)
+        logger.log(DEBUG, "FragmentLifecycleIntegration removed.")
+    }
+
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        (activity as? FragmentActivity)
+            ?.supportFragmentManager
+            ?.registerFragmentLifecycleCallbacks(
+                SentryFragmentLifecycleCallbacks(hub),
```

Comment:
This creates a new instance of `SentryFragmentLifecycleCallbacks`. Another way would be to create `SentryFragmentLifecycleCallbacks` and reuse them for all activities. I'm not sure which one is preferred, WDYT?

Reply:
I don't know the cost at runtime using one or the other, there's no static field so either as a singleton or a new instance will behave exactly the same, I'd keep creating a new instance though, because it receives a `hub` as a param, so it could be used standalone with a custom `hub`